src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -441,6 +445,11 @@ internal SingleArgumentJumpThunk JumpThunk(ExternSymbolNode target, ISymbolNode
 
         public IMethodNode MethodEntrypoint(MethodDesc method, bool unboxingStub = false)
         {
+            if (!_cppCodeGen)
```

Comment:
Should the shared generics be disabled for cppcodegen for now by initializing the type system without shared generics instead?